File: cpp/011_ContainerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        int max = 0;
        while(left < right){
            int t = std::min(height[left], height[right]);
            max = std::max(max, t * (right - left));
            while(height[left] <= t && left < right) left++;
            while(height[right] <= t && left < right) right--;
        }
        return max;
    }
};

File: js/011_ContainerWithMostWater.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var left = 0;
    var right = height.length-1;
    var max = 0;
    while(left < right){
        var h = Math.min(height[left], height[right]);
        max = Math.max(max, h * (right-left));
        while(height[left] <= h && left < right) left++;
        while(height[right] <= h && left < right) right--;
    }
    return max;
};